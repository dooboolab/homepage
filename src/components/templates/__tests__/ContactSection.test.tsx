import 'react-native';

import * as React from 'react';

import {RenderAPI, render} from '@testing-library/react-native';
import {createTestElement, createTestProps} from '../../../../test/testUtils';

import Shared from '../ContactSection';

let props: any;
let component: React.ReactElement;
let testingLib: RenderAPI;

jest.mock('@react-navigation/native', () => {
  return {
    // @ts-ignore
    ...jest.requireActual('@react-navigation/native'),
    createNavigatorFactory: jest.fn(),
    useNavigation: (): Record<string, unknown> => ({
      navigate: jest.fn(),
      setOptions: jest.fn(),
    }),
  };
});

describe('Rendering', () => {
  beforeEach(() => {
    props = createTestProps();
    component = createTestElement(<Shared {...props} />);
    testingLib = render(component);
  });

  it('renders without crashing', () => {
    const baseElement = testingLib.toJSON();

    expect(baseElement).toMatchSnapshot();
    expect(baseElement).toBeTruthy();
  });
});

describe('Interaction', () => {
  beforeEach(() => {
    testingLib = render(component);
  });

  it('should simulate onClick', () => {
    expect(testingLib.toJSON()).toMatchSnapshot();
    // const btn = testingLib.queryByTestId('btn');
    // act(() => {
    //   fireEvent.press(btn);
    // });
    // expect(cnt).toBe(3);
  });
});
