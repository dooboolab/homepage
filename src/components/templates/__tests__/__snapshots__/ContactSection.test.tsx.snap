// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interaction should simulate onClick 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#FFFFFF",
        "minHeight": 600,
      },
    ]
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingBottom": 0,
          "paddingLeft": 24,
          "paddingRight": 24,
          "paddingTop": 0,
        },
      ]
    }
  >
    <Image
      resizeMode="cover"
      source={
        Object {
          "createTransformer": [Function],
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          Object {
            "opacity": 0.65,
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "flexDirection": "column",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "fontFamily": "futura",
              "fontSize": 30,
              "fontWeight": "500",
              "maxWidth": "60%",
              "textAlign": "center",
            },
          ]
        }
      >
        Who's next? Feel free to talk.
      </Text>
      <View
        style={
          Array [
            Object {
              "alignSelf": "center",
              "flexDirection": "column",
              "marginTop": 56,
              "width": "86%",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Please write your name"
          placeholderTextColor="#999999"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "#D3D8E8",
                "borderColor": "#393D7A",
                "borderRadius": 10,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 14,
                "marginBottom": 12,
                "opacity": 0.6,
                "paddingBottom": 16,
                "paddingLeft": 18,
                "paddingRight": 18,
                "paddingTop": 16,
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="email@email.com"
          placeholderTextColor="#999999"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "#D3D8E8",
                "borderColor": "#393D7A",
                "borderRadius": 10,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 14,
                "marginBottom": 12,
                "opacity": 0.6,
                "paddingBottom": 16,
                "paddingLeft": 18,
                "paddingRight": 18,
                "paddingTop": 16,
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <TextInput
          allowFontScaling={true}
          multiline={true}
          numberOfLines={10}
          onChangeText={[Function]}
          placeholder="Please tell us your stories"
          placeholderTextColor="#999999"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "#D3D8E8",
                "borderColor": "#393D7A",
                "borderRadius": 10,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 14,
                "marginBottom": 12,
                "opacity": 0.6,
                "paddingBottom": 16,
                "paddingLeft": 18,
                "paddingRight": 18,
                "paddingTop": 16,
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          activeOpacity={0.7}
          delayPressIn={50}
          onPress={[MockFunction]}
          style={
            Object {
              "marginTop": 36,
            }
          }
        >
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "backgroundColor": "#393D7A",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingBottom": 8,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 8,
                },
                Object {
                  "backgroundColor": "#00BA90",
                  "borderRadius": 20,
                  "height": 44,
                  "paddingHorizontal": 40,
                },
                false,
                undefined,
              ]
            }
            testID="button-view"
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 14,
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "avenir",
                    "fontSize": 16,
                  },
                  undefined,
                ]
              }
            >
              Send
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Rendering renders without crashing 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#FFFFFF",
        "minHeight": 600,
      },
    ]
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexBasis": 0,
          "flexDirection": "column",
          "flexGrow": 1,
          "flexShrink": 1,
          "justifyContent": "center",
          "paddingBottom": 0,
          "paddingLeft": 24,
          "paddingRight": 24,
          "paddingTop": 0,
        },
      ]
    }
  >
    <Image
      resizeMode="cover"
      source={
        Object {
          "createTransformer": [Function],
        }
      }
      style={
        Array [
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          Object {
            "height": undefined,
            "width": undefined,
          },
          Object {
            "opacity": 0.65,
          },
        ]
      }
    />
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "flexDirection": "column",
            "justifyContent": "center",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#000000",
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "fontFamily": "futura",
              "fontSize": 30,
              "fontWeight": "500",
              "maxWidth": "60%",
              "textAlign": "center",
            },
          ]
        }
      >
        Who's next? Feel free to talk.
      </Text>
      <View
        style={
          Array [
            Object {
              "alignSelf": "center",
              "flexDirection": "column",
              "marginTop": 56,
              "width": "86%",
            },
          ]
        }
      >
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="Please write your name"
          placeholderTextColor="#999999"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "#D3D8E8",
                "borderColor": "#393D7A",
                "borderRadius": 10,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 14,
                "marginBottom": 12,
                "opacity": 0.6,
                "paddingBottom": 16,
                "paddingLeft": 18,
                "paddingRight": 18,
                "paddingTop": 16,
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholder="email@email.com"
          placeholderTextColor="#999999"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "#D3D8E8",
                "borderColor": "#393D7A",
                "borderRadius": 10,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 14,
                "marginBottom": 12,
                "opacity": 0.6,
                "paddingBottom": 16,
                "paddingLeft": 18,
                "paddingRight": 18,
                "paddingTop": 16,
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <TextInput
          allowFontScaling={true}
          multiline={true}
          numberOfLines={10}
          onChangeText={[Function]}
          placeholder="Please tell us your stories"
          placeholderTextColor="#999999"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "alignSelf": "stretch",
                "backgroundColor": "#D3D8E8",
                "borderColor": "#393D7A",
                "borderRadius": 10,
                "borderWidth": 1,
                "color": "#000000",
                "fontSize": 14,
                "marginBottom": 12,
                "opacity": 0.6,
                "paddingBottom": 16,
                "paddingLeft": 18,
                "paddingRight": 18,
                "paddingTop": 16,
              },
            ]
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          activeOpacity={0.7}
          delayPressIn={50}
          onPress={[MockFunction]}
          style={
            Object {
              "marginTop": 36,
            }
          }
        >
          <View
            onLayout={[Function]}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "alignSelf": "stretch",
                  "backgroundColor": "#393D7A",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingBottom": 8,
                  "paddingLeft": 12,
                  "paddingRight": 12,
                  "paddingTop": 8,
                },
                Object {
                  "backgroundColor": "#00BA90",
                  "borderRadius": 20,
                  "height": 44,
                  "paddingHorizontal": 40,
                },
                false,
                undefined,
              ]
            }
            testID="button-view"
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "white",
                    "fontSize": 14,
                  },
                  Object {
                    "color": "#000000",
                    "fontFamily": "avenir",
                    "fontSize": 16,
                  },
                  undefined,
                ]
              }
            >
              Send
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
